codec-http/src/main/java/io/netty/handler/codec/http/websocketx/WebSocketClientProtocolConfig.java

```diff
@@ -0,0 +1,329 @@
+            this.webSocketUri = webSocketUri;
+            return this;
+        }
+
+        /**
+         * Sub protocol request sent to the server.
+         */
+        public Builder subprotocol(String subprotocol) {
+            this.subprotocol = subprotocol;
+            return this;
+        }
+
+        /**
+         * Version of web socket specification to use to connect to the server
+         */
+        public Builder version(WebSocketVersion version) {
+            this.version = version;
+            return this;
+        }
+
+        /**
+         * Allow extensions to be used in the reserved bits of the web socket frame
+         */
+        public Builder allowExtensions(boolean allowExtensions) {
+            this.allowExtensions = allowExtensions;
+            return this;
+        }
+
+        /**
+         * Map of custom headers to add to the client request
+         */
+        public Builder customHeaders(HttpHeaders customHeaders) {
+            this.customHeaders = customHeaders;
+            return this;
+        }
+
+        /**
+         * Maximum length of a frame's payload
+         */
+        public Builder maxFramePayloadLength(int maxFramePayloadLength) {
+            this.maxFramePayloadLength = maxFramePayloadLength;
+            return this;
+        }
+
+        /**
+         * Whether to mask all written websocket frames. This must be set to true in order to be fully compatible
+         * with the websocket specifications. Client applications that communicate with a non-standard server
+         * which doesn't require masking might set this to false to achieve a higher performance.
+         */
+        public Builder performMasking(boolean performMasking) {
+            this.performMasking = performMasking;
+            return this;
+        }
+
+        /**
+         * When set to true, frames which are not masked properly according to the standard will still be accepted.
+         */
+        public Builder allowMaskMismatch(boolean allowMaskMismatch) {
+            this.allowMaskMismatch = allowMaskMismatch;
+            return this;
+        }
+
+        /**
+         * {@code true} if close frames should not be forwarded and just close the channel
+         */
+        public Builder handleCloseFrames(boolean handleCloseFrames) {
+            this.handleCloseFrames = handleCloseFrames;
+            return this;
+        }
+
+        /**
+         * {@code true} if pong frames should not be forwarded
+         */
+        public Builder dropPongFrames(boolean dropPongFrames) {
+            this.dropPongFrames = dropPongFrames;
+            return this;
+        }
+
+        /**
+         * Handshake timeout in mills, when handshake timeout, will trigger user
+         * event {@link ClientHandshakeStateEvent#HANDSHAKE_TIMEOUT}
+         */
+        public Builder handshakeTimeoutMillis(long handshakeTimeoutMillis) {
+            this.handshakeTimeoutMillis = handshakeTimeoutMillis;
```

Comment:
Should we have also the validation here as we do in: https://github.com/netty/netty/pull/9699/files#diff-b95c7e6a26fdf04d7651f3629be351d1R76

Reply:
I'd rather prefer to move all validations into ```build``` phase.